prov/sockets/src/sock_progress.c

```diff
@@ -2464,6 +2463,11 @@ static void sock_thread_set_affinity(char *s)
 	int j, first, last,stride;
 	cpu_set_t mycpuset;
 	pthread_t mythread;
+ 	
+	if(!s) {
+		SOCK_LOG_DBG("Invalid FI_SOCKETS_PE_AFFINITY value\n");
+		return;
+	}
```

Comment:
This check shouldn't be needed.  This comes from fi_var_get_str().  If that call returns success, the return string will not be NULL.